poc/events.py

```diff
@@ -0,0 +1,63 @@
+"""
+Event data types for the Message-Bus/Event-Driven Architecture
+
+    - Makes it easier to create multiple listeners for different events
+"""
+from dataclasses import dataclass
+from datetime import datetime
+from typing import Any, Dict
+
+
+class Event:
+    """
+    Base class for all events
+
+        - Implements some useful helper functions
+        - Useful for exception handling/type hints
+    """
+
+    @classmethod
+    def from_dict(cls, data: Dict[Any, Any]):
```

Comment:
```suggestion
    def from_dict(cls, data: Dict[str, Any]):
```
field names are always string